lib/shortenator.rb

```diff
@@ -52,6 +57,10 @@ def validate_config
       unless Integer === config.retry_amount && config.retry_amount >= 0
         raise Error, "retry amount must be a number equal or greater than 0, saw #{config.retry_amount}"
       end
+
+      unless caching_model.nil?
+        validate_caching_model
+      end
```

Comment:
Outside of the scope of the PR, but I wonder if configuration validation shouldn't happen at configuration time, rather than potentially throwing a runtime error at the point that a link is being shortened.

Reply:
I really like this suggestion since this allows the gem to still work even though the model isn't valid. changes were made.